### PR TITLE
Update boolean configuration cluster XML according to latest spec

### LINKED ISSUE #30897: [1.3] Valve and Boolean Sensor renamed in latest spec
The valve device type has been renamed to Water Valve in the latest spec, we need to align the SDK with this latest change.

The Boolean Sensor Configuration cluster has been renamed to Boolean State Configuration, we need to align the SDK with this latest change.

Spec PR's
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8649
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8648

### PR DESCRIPTION
**Pull request by fessehaeve**
keeps XML up to date with spec fixes #30897

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/boolean-sensor-configuration-cluster.xml [hunk 1]
```diff
@@ -31,12 +31,10 @@ limitations under the License.
...
     <cluster code="0x0080"/>
-    <item name="High" value="0x0"/>
-    <item name="Standard" value="0x1"/>
-    <item name="Low" value="0x2"/>
-  </enum>
+    <field name="GneralFault" mask="0x1"/>
+  </bitmap>
 
   <cluster apiMaturity="provisional">
     <name>Boolean Sensor Configuration</name>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/boolean-sensor-configuration-cluster.xml:40]: In my spec copy the name of this is `Boolean State Configuration Cluster`.

We should either match the name here or update the spec to use this name.

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
Skip protocol compatibility: boolean sensor config is a provisional cluster.

File: src/app/zap-templates/zcl/data-model/chip/boolean-state-configuration-cluster.xml [hunk 2]
```diff
@@ -31,41 +31,49 @@ limitations under the License.
...
   </bitmap>
 
-  <enum name="SensitivityEnum" type="enum8">
+  <bitmap name="SensorFaultBitmap" type="bitmap16">
     <cluster code="0x0080"/>
-    <item name="High" value="0x0"/>
-    <item name="Standard" value="0x1"/>
-    <item name="Low" value="0x2"/>
-  </enum>
+    <field name="GneralFault" mask="0x1"/>
```
Comment (ReneJosefsen) [inline src/app/zap-templates/zcl/data-model/chip/boolean-state-configuration-cluster.xml:36]:
```suggestion
    <field name="GeneralFault" mask="0x1"/>
```

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 3]
```diff
@@ -2083,19 +2083,18 @@ limitations under the License.
...
         <deviceId editable="false">0x0041</deviceId>
         <class>Simple</class>
         <scope>Endpoint</scope>
         <clusters lockOthers="true">
             <include cluster="Identify" client="false" server="true" clientLocked="true" serverLocked="true"></include>
             <include cluster="Boolean State" client="false" server="true" clientLocked="true" serverLocked="true"></include>
-            <include cluster="Descriptor" client="false" server="true" clientLocked="true" serverLocked="true">
-                <requireAttribute>TAG_LIST</requireAttribute>
-            </include>
+            <include cluster="Boolean State Configuration" client="false" server="true" clientLocked="true" serverLocked="true"></include>
```
Comment (ReneJosefsen) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2096]: I might be wrong here, does serverLocked indicate if this is mandated or not, in that case, this should be false?

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 4]
```diff
@@ -2083,19 +2083,18 @@ limitations under the License.
         </clusters>
     </deviceType>
     <deviceType>
-        <name>MA-boolean-sensor</name>
+        <name>MA-water-freeze-detector</name>
```
Comment (ReneJosefsen) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2086]: It looks like we are missing the rain sensor here?

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 5]
```diff
@@ -2120,14 +2119,11 @@ limitations under the License.
...
         <scope>Endpoint</scope>
-        <superset>Matter Boolean Sensor</superset>
         <clusters lockOthers="true">
             <include cluster="Identify" client="false" server="true" clientLocked="true" serverLocked="true"></include>
             <include cluster="Boolean State" client="false" server="true" clientLocked="true" serverLocked="true"></include>
-            <include cluster="Boolean Sensor Configuration" client="false" server="true" clientLocked="true" serverLocked="true"></include>
-            <include cluster="Descriptor" client="false" server="true" clientLocked="true" serverLocked="true">
-                <requireAttribute>TAG_LIST</requireAttribute>
-            </include>
+            <include cluster="Boolean State Configuration" client="false" server="true" clientLocked="true" serverLocked="true"></include>
```
Comment (ReneJosefsen) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2125]: Same as above, should the serverLocked tag be true?

File: src/controller/data_model/controller-clusters.matter [hunk 6]
```diff
@@ -4018,31 +4012,46 @@ provisional cluster BooleanSensorConfiguration = 128 {
     kSensitivityLevel = 0x8;
   }
 
+  bitmap SensorFaultBitmap : bitmap16 {
+    kGneralFault = 0x1;
```
Comment (ReneJosefsen) [inline src/controller/data_model/controller-clusters.matter:4016]: I guess this is generated, so if we fix the XML this is also fixed?

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 4 above)
Comment (fessehaeve) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2086]: Missed it. Will include it, thanks.

File: src/controller/data_model/controller-clusters.matter (hunk 6 above)
Comment (fessehaeve) [reply src/controller/data_model/controller-clusters.matter:4016]: Yes, will be updated.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 3 above)
Comment (fessehaeve) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2096]: You are right. Will update it.

Comment (fessehaeve) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:2096]: I see all mixed use of these properties. I don't think everyone understands how to use these.

**Review by ReneJosefsen — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**